plugin/core/windows.py

```diff
@@ -316,6 +316,7 @@ def __init__(
         self._workspace = workspace
         self._workspace.on_changed = self._on_project_changed
         self._workspace.on_switched = self._on_project_switched
+        self._progress = dict()  # type: Dict[Any, Any]
```

Comment:
There is also this case to handle:
> In case the client sets up a progress token in the initialize params (e.g. property workDoneToken) the server is also allowed to use that token (and only that token) using the $/progress notification sent from the server to the client.